pkg/services/govmomi/service.go

```diff
@@ -218,30 +220,51 @@ func (vms *VMService) DestroyVM(ctx *context.VMContext) (infrav1.VirtualMachine,
 		State:     &vm,
 	}
 
-	// Power off the VM.
+	// Shut down the VM
 	powerState, err := vms.getPowerState(vmCtx)
 	if err != nil {
-		return vm, err
+		return reconcile.Result{}, vm, err
 	}
+
 	if powerState == infrav1.VirtualMachinePowerStatePoweredOn {
-		task, err := vmCtx.Obj.PowerOff(ctx)
+		// Trigger the soft power off and set the condition.
+		softPowerOffPending, err := vms.triggerSoftPowerOff(vmCtx)
 		if err != nil {
-			return vm, err
+			return reconcile.Result{}, vm, err
 		}
-		ctx.VSphereVM.Status.TaskRef = task.Reference().Value
-		if err = ctx.Patch(); err != nil {
-			ctx.Logger.Error(err, "patch failed", "vm", ctx.String())
-			return vm, err
+
+		if softPowerOffPending {
+			// Return to reconcile later when we have a pending soft power off operation.
+			return reconcile.Result{RequeueAfter: time.Minute}, vm, nil
 		}
-		ctx.Logger.Info("wait for VM to be powered off")
-		return vm, nil
+
+		// Hard shut off VM.
+		task, err := vmCtx.Obj.PowerOff(vmCtx)
```

Comment:
```suggestion
		task, err := vmCtx.Obj.PowerOff(ctx)
```
nit: Let's use the normal context whenever possible (like here before this change)

Reply:
Is done, can be resolved